gulpfile.js

```diff
@@ -0,0 +1,51 @@
+const fs = require('fs');
+const captureWebsite = require('capture-website');
+
+
+const themesDir = __dirname + '/themes';
+const screenshotDir = __dirname + '/screenshots';
+
+function getThemes() {
+	return fs.readdirSync(themesDir).map(f => (/^.+(?=\.css$)/.exec(f) || [''])[0]).filter(f => f);
+}
+
+/**
+ * Takes a screenshot of all themes overwriting the old ones.
+ */
+async function screenshotAllThemes() {
+	for (const theme of getThemes()) {
+		await screenshotTheme(theme, true);
+	}
+}
+
+/**
+ * Takes a screenshot of themes which don't have one already.
+ */
+async function screenshotMissingThemes() {
+	for (const theme of getThemes()) {
+		await screenshotTheme(theme, false);
+	}
+}
+
+async function screenshotTheme(theme, overwrite) {
+	const file = `${screenshotDir}/${theme}.png`;
+	if (fs.existsSync(file)) {
+		if (overwrite) {
+			fs.unlinkSync(file);
+		} else {
+			return;
+		}
+	}
```

Comment:
We can improve the async-ness of these tasks a bit if we use `fs.promises` API. It's technically experimental: https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_promises_api but it would allow us to interleave this is a little bit better.

Besides switching the `existsSync` & `unlinkSync` to `await exists` etc., we would also need to switch the `for..of` loops to `getThemes().map(...)` and await a `Promise.all` of all the screenshot promises, which should allow them to run in parallel.

Reply:
Agreed.

> which should allow them to run in parallel

It's intentionally not parallel because this would cause a memory leak in the underlying library. I'll add a comment.